app/models/spree/flow_io_credit_card_decorator.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Spree
+  module FlowIoCreditCardDecorator
+    def self.prepended(base)
+      base.serialize :meta, ActiveRecord::Coders::JSON.new(symbolize_keys: true)
+
+      base.store_accessor :meta, :flow_data
+    end
+
+    def push_authorization(auth_hash)
+      self.flow_data ||= {}
+      flow_data['authorizations'] ||= []
+      card_authorizations = flow_data['authorizations']
+      card_authorizations.delete_if { |ca| ca['id'] == auth_hash['id'] }
+      card_authorizations << auth_hash
+    end
+
```

Comment:
@texpert Are we using Spree::CreditCard modeling for handling payments? If it generates problem we. could create a new payment method dedicated to flow

Reply:
We're not handling payments, we're just mapping the received info from flow.io into Spree, so everything will continue to be flawlessly processed - displaying of payments and captures in Spree::Admin, refunding operations, anything.